src/network/networkd-link.c

```diff
@@ -794,6 +794,8 @@ static int link_set_routing_policy_rule(Link *link) {
         assert(link);
         assert(link->network);
 
+        link->routing_policy_rules_configured = false;
```

Comment:
These should all be set to false at the same location in the codebase, likely when networkd determines that we need to reconfigure a link. Otherwise the invariant for link_check_ready() might not hold.